### ISSUE TITLE
Remove PheWAS invalid evidence strings where the target is blank

### ISSUE THREAD

**Issue by ireneisdoomed** [opentargets/platform#1494]
Investigating the output of the latest 21.04 pipeline run, @d0choa has identified that there are 90 invalid evidence strings for PheWAS where `targetId == ""`.

This is because the PheWAS parser executes the step of mapping the targets to the corresponding ensembl ID using the [HGNCParser.py](https://github.com/opentargets/evidence_datasource_parsers/blob/master/common/HGNCParser.py) tool. 
All those that are not mappable (the result of the mapping is "NaN") are discarded (see [here](https://github.com/opentargets/evidence_datasource_parsers/blob/02f4bcfd797abd59fd1112018503e99b423f7cc9/modules/PheWAS.py#L93.)). The ETL takes as input for the `targetId` field the value present in `targetFromSourceId`.

In this case the target is not mappable because it is a Uniprot ID of a non-human species and therefore should not be in the evidence in the first place. We can either:
- Add an ad-hoc check in the parser to filter out the cases where `targetFromSourceId == ""`.
- Change the HGNCParser util to handle these cases where the inputted ID is not present in HGNC. I would go for this one, as it is the more correct approach. Although we should also bare in mind that currently PheWAS is the only parser using this util.

**Comment by d0choa** [comment]
why not leaving the ETL handle the target mappings?

**Comment by ireneisdoomed** [comment]
Currently we need to parse the targets in PheWAS in order to be able to combine the data with the variant index.

As you say, we could do this step again in the ETL by taking the symbol in `targetFromSource` and not the previously computed ENSID in `targetFromSourceId`.

However it seems to me that the action is needed in the evidence generation step as we know we are providing as evidence something that we are not going to be able to capture.

**Comment by ireneisdoomed** [comment]
This issue is now solved with the PR [#95](https://github.com/opentargets/evidence_datasource_parsers/pull/95).

The implementation does not consist of any of the solutions mentioned above. We went to the root cause of the problem and avoided the necessity of parsing the genes to ENSIDs (more details in the PR).

I've collected the code to extract the variant data and consequences for the genes present in PheWAS in this gist, for us to be able to reproduce it in the future.

https://gist.github.com/ireneisdoomed/635cb1c6327f9d2f0a05fc561d3cce2c
